File: bindings/dotnet/DstuCore/Selftest.cs
using DstuCore.Native;

namespace DstuCore;

/// <summary>Runtime known-answer self-check (T-161/D-117) - lets a consumer verify their exact
/// installed native library is producing correct outputs on their exact platform, without
/// trusting "it compiled" alone.</summary>
public static class Selftest
{
    /// <summary>Re-verifies one official test vector per primitive (Kalyna, Kupyna, Strumok, DSTU
    /// 4145) against the live compiled <c>dstu_core_capi</c> native library. Throws
    /// <see cref="DstuException"/> if any primitive fails.</summary>
    public static void Run()
    {
        NativeStatus.ThrowIfError(NativeMethods.dstu_selftest());
    }
}

File: bindings/java/src/main/java/ua/dstucrypto/dstucore/Selftest.java
package ua.dstucrypto.dstucore;

/**
 * Re-runs dstu_core's official-vector self-check against this exact compiled native build.
 */
public final class Selftest {
    static {
        NativeLoader.ensureLoaded();
    }

    private Selftest() {
    }

    /**
     * Throws {@link DstuException} naming which primitive(s) failed, if any.
     */
    public static native void run();
}
